### PR TITLE
Implement Headless SwapChain mode.

### PR DESCRIPTION
**Pull request by pixelflinger**
This is useful for unit tests. The content of the swapchain can be
read to main memory with Renderer::readPixels().

This PR doesn't implement this new feature in the following backends and
platforms (TODO):

VulkanDriver.cpp
MetalDriver.mm

PlatformWGL.cpp
PlatformWebGL.cpp
PlatformCocoaGL.mm
PlatformCocoaTouchGL.mm

### CODE REVIEW

File: filament/backend/src/opengl/PlatformEGL.cpp [hunk 1]
```diff
@@ -361,6 +363,26 @@ Platform::SwapChain* PlatformEGL::createSwapChain(
...
+            EGL_HEIGHT, EGLint(height),
+            EGL_NONE
+    };
+
+    EGLSurface sur = eglCreatePbufferSurface(mEGLDisplay,
+                (flags & backend::SWAP_CHAIN_CONFIG_TRANSPARENT) ?
+                mEGLTransparentConfig : mEGLConfig, attribs);
+
+    if (UTILS_UNLIKELY(sur == EGL_NO_SURFACE)) {
+        logEglError("eglCreateWindowSurface");
```
Comment (prideout) [inline filament/backend/src/opengl/PlatformEGL.cpp]: Error message should be `eglCreatePbufferSurface`

File: filament/backend/src/opengl/PlatformGLX.cpp [hunk 2]
```diff
@@ -238,15 +238,34 @@ void PlatformGLX::terminate() noexcept {
...
+        mPBuffers.push_back(sur);
+    }
+    return (SwapChain*) sur;
+}
+
+void PlatformGLX::destroySwapChain(Platform::SwapChain* swapChain) noexcept {
+    auto it = std::find(mPBuffer.begin(), mPBuffer.end(), (GLXPbuffer)swapChain);
+    if (it != mPBuffer.end()) {
+        g_glx.destroyPbuffer(mGLXDisplay, (GLXPbuffer)swapChain);
+        mPBuffer.erase(it);
```
Comment (bejado) [inline filament/backend/src/opengl/PlatformGLX.cpp]: `mPBuffer` -> `mPBuffers`.

File: filament/backend/src/opengl/PlatformGLX.h [hunk 3]
```diff
@@ -36,6 +38,7 @@ class PlatformGLX final : public backend::OpenGLPlatform {
     void terminate() noexcept override;
 
     SwapChain* createSwapChain(void* nativewindow, uint64_t& flags) noexcept override;
+    SwapChain* createSwapChain(uint32_t width, uint32_t height, uint64_t& flags) noexcept override;
```
Comment (bejado) [inline filament/backend/src/opengl/PlatformGLX.h]: Use `final` for consistency with other platforms? (in fact, all of these methods could probably be made `final`)

File: filament/include/filament/Engine.h [hunk 4]
```diff
@@ -259,6 +259,20 @@ class UTILS_PUBLIC Engine {
      */
     SwapChain* createSwapChain(void* nativeWindow, uint64_t flags = 0) noexcept;
 
+
+    /**
+     * Creates a headless SwapChain.
+     *
+      * @param width    Width of the drawing buffer in pixels.
+      * @param height   Height of the drawing buffer in pixels.
```
Comment (bejado) [inline filament/include/filament/Engine.h]: nit: extra space

**Pull request merged**